tfjs-backend-webgpu/src/kernels/Softmax.ts

```diff
@@ -32,35 +29,22 @@ export function softmax(
   const {logits} = inputs;
   const {dim} = attrs;
 
-  const axes = util.parseAxisParam([dim], logits.shape);
-
-  const maxLogit = max({
+  const logitsReshaped = reshape({
     inputs: {x: logits},
     backend,
-    attrs: {reductionIndices: axes, keepDims: false}
+    attrs: {
+      shape: [
+        util.sizeFromShape(logits.shape) / logits.shape[dim], logits.shape[dim]
+      ]
+    }
   });
-
-  const expandedShape = backend_util.expandShapeToKeepDim(maxLogit.shape, axes);
-
-  const maxLogitsReshaped =
-      reshape({inputs: {x: maxLogit}, backend, attrs: {shape: expandedShape}});
-  const program = new SoftmaxProgram(logits.shape);
-  const b = backend.runWebGPUProgram(
-      program, [logits, maxLogitsReshaped], logits.dtype);
-  const sumExp =
-      sum({inputs: {x: b}, backend, attrs: {axis: axes, keepDims: false}});
-  const sumExpReshaped =
-      reshape({inputs: {x: sumExp}, backend, attrs: {shape: expandedShape}});
-  const res =
-      realDiv({inputs: {a: b, b: sumExpReshaped}, backend}) as TensorInfo;
-
-  backend.disposeData(maxLogit.dataId);
-  backend.disposeData(maxLogitsReshaped.dataId);
-  backend.disposeData(b.dataId);
-  backend.disposeData(sumExp.dataId);
-  backend.disposeData(sumExpReshaped.dataId);
-
-  return res;
+  const program = new SoftmaxProgram(logitsReshaped.shape);
+  const res = backend.runWebGPUProgram(program, [logitsReshaped], logits.dtype);
```

Comment:
I think the result elements' type is float, and (0, 1], if the dtype of `logits` is int, will there be an error?